Cabal/src/Distribution/Simple/GHC/Build/ExtraSources.hs

```diff
@@ -73,7 +74,15 @@ buildCSources
 buildCSources mbMainFile =
   buildExtraSources
     "C Sources"
-    Internal.componentCcGhcOptions
+    ( \verbosity lbi bi clbi odir filename ->
+        (Internal.sourcesGhcOptions verbosity lbi bi clbi odir filename)
+          { -- C++ compiler options: GHC >= 8.10 requires -optcxx, older requires -optc
+            -- we want to be able to support cxx-options and cc-options separately
+            ghcOptCxxOptions = Internal.separateGhcOptions (mkVersion [8, 10]) (compiler lbi) (Internal.defaultGhcOptCxxOptions lbi bi)
+          , -- there are problems with linking with versions lower than 9.4, that's why we need this duplication
```

Comment:
what "duplication"? With the code for `buildCxxOptions` below? Please, mention it

Reply:
I need to change the description, then the line is needed to re-assign `ghcOptCcProgram`